ecs-cli/modules/cli/cluster/cluster_app.go

```diff
@@ -282,22 +282,17 @@ func createCluster(context *cli.Context, awsClients *AWSClients, commandConfig *
 	}
 
 	if launchType == config.LaunchTypeEC2 {
-		architecture := context.String(flags.ArchitectureFlag)
-		if err := validateArchitecture(architecture); err != nil {
-			return err
-		}
+		architecture := amimetadata.ArchitectureTypeX86
 
-		if architecture == amimetadata.ArchitectureTypeARM64 {
-			instanceTypeParam, err := cfnParams.GetParameter(ParameterKeyInstanceType)
-			// arm64 requires the a1 instance family
-			if err == cloudformation.ParameterNotFoundError {
-				logrus.Infof("Defaulting instance type to %s because architecture was %s", defaultARM64InstanceType, amimetadata.ArchitectureTypeARM64)
-				cfnParams.Add(ParameterKeyInstanceType, defaultARM64InstanceType)
-			} else if err != nil {
-				return err
-			} else if !strings.HasPrefix(aws.StringValue(instanceTypeParam.ParameterValue), "a1") {
-				return fmt.Errorf("%s requires the a1 instance family", amimetadata.ArchitectureTypeARM64)
-			}
+		// a1 instances get the Arm based ECS AMI
+		instanceTypeParam, err := cfnParams.GetParameter(ParameterKeyInstanceType)
+		if err == cloudformation.ParameterNotFoundError {
+			logrus.Infof("Defaulting instance type to t2.micro")
+		} else if err != nil {
+			return err
+		} else if instanceType := aws.StringValue(instanceTypeParam.ParameterValue); strings.HasPrefix(instanceType, "a1") {
```

Comment:
I'm wondering if we could maybe use a regular expression or match in a list of instance types we know are ARM64 here instead. The `HasPrefix` method is maybe too simple, since you could specify "a1fooobar" -- of course this would fail later since it's not a supported instance type in our template.go, but might be both more robust and explicit if we either keep a list or use regex matching.

Reply:
I like the idea of using a regex; this works for everything they have right now and should match anything they might add in the future: https://regexr.com/464kr